File: we_9_5_huber_down_weighting_of_an_outlier.py
# Integrated Navigation Systems — Chapter 9: Factor Graphs, Nonlinear Optimization, and AI-Aided Estimation
# Worked Example 9.5 — Huber down-weighting of an outlier
# Companion code (educational reference implementation; see the book for context).

e=4.0; delta=1.345
w = 1.0 if abs(e)<=delta else delta/abs(e)
print(f'Huber weight={w:.3f}')   # 0.336

File: we_9_5_huber_down_weighting_of_an_outlier.m
% Integrated Navigation Systems — Chapter 9: Factor Graphs, Nonlinear Optimization, and AI-Aided Estimation
% Worked Example 9.5 — Huber down-weighting of an outlier
% Companion code (educational reference implementation; see the book for context).

e=4.0; delta=1.345;
w = (abs(e)<=delta)*1 + (abs(e)>delta)*delta/abs(e);
fprintf('Huber weight=%.3f\n', w);   % 0.336
